src/array_api_extra/_lib/_at.py

```diff
@@ -298,7 +300,7 @@ def _op(
             and idx.dtype == xp.bool
             and idx.shape == x.shape
         ):
-            y_xp = xp.asarray(y, dtype=x.dtype)
+            y_xp = xp.asarray(y, dtype=x.dtype, device=_compat.device(x))
```

Comment:
Untested fix, which only benefits eager JAX.
On jax.jit, device propagation fails due to https://github.com/jax-ml/jax/issues/26000

In a follow-up PR I'll rework the `test_device` tests to align them to the pattern recently established in scipy.